### PR TITLE
Add sign-on-methods-wrapper component

### PR DESCRIPTION
**Pull request by NutharaNR**
### Purpose
Purpose is to restructure the component structure in sign-on-methods by adding sign-on-methods-wrapper.tsx.

This re-structuring is intended to do the modifications required to add AILoginFLowProvider in the sign-on-methods.tsx as follows:

```tsx
return (
    <AILoginFlowProvider>
          <SignOnMethodsWrapper
              application={ application }
              appId={ appId }
              authenticationSequence={ authenticationSequence }
              clientId={ clientId }
              isLoading={ isLoading }
              onUpdate={ onUpdate }
              readOnly={ readOnly }
              isSystemApplication={ isSystemApplication }
              hiddenAuthenticators={ hiddenAuthenticators }
              data-componentid={ componentId }
          />
    </ AILoginFlowProvider>
);
```
sign-on-methods-wrapper.tsx will handle the modifications for the parameters that will be passed to sign-on-methods-core.tsx.

### Related Issues
- 

### Related PRs
- 

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: .changeset/odd-mirrors-check.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@wso2is/features": patch
```
Comment (brionmario) [inline .changeset/odd-mirrors-check.md:2]:
```suggestion
"@wso2is/console": patch
"@wso2is/features": patch
```

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+     * Flag to determine if the updated application a system application.
+     */
+    isSystemApplication?: boolean;
+    /**
+     * List of hidden authenticators.
+     */
+    hiddenAuthenticators: string[];
+}
+
+
```
Comment (brionmario) [inline features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:71]:
```suggestion
```

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            appId={ appId }
+            authenticationSequence={ authenticationSequence }
+            clientId={ clientId }
+            isLoading={ isLoading }
+            onUpdate={ onUpdate }
+            readOnly={ readOnly }
+            isSystemApplication={ isSystemApplication }
+            hiddenAuthenticators={ hiddenAuthenticators }
+            data-componentid={ componentId  }
+
```
Comment (brionmario) [inline features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:109]: Remove the extra spaces.

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        authenticationSequence,
+        clientId,
+        isLoading,
+        onUpdate,
+        readOnly,
+        isSystemApplication,
+        hiddenAuthenticators,
+        [ "data-componentid" ]: componentId
+    } = props;
+
```
Comment (brionmario) [inline features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:95]: Remove the line breaks.

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx (hunk 2 above)
Comment (brionmario) [reply features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:71]: Remove the line break.

File: features/admin.applications.v1/components/settings/sign-on-methods/sign-on-methods.tsx [hunk 5]
```diff
@@ -110,10 +110,3 @@ export const SignOnMethods: FunctionComponent<SignOnMethodsPropsInterface> = (
         />
     );
 };
-
-/**
- * Default props for the application sign-on-methods component.
- */
-SignOnMethods.defaultProps = {
```
Comment (brionmario) [inline features/admin.applications.v1/components/settings/sign-on-methods/sign-on-methods.tsx:117]: Verify if removing the `componentid` cause any test suite failures.

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx (hunk 3 above)
Comment (NutharaNR) [reply features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:109]: Removed.

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx (hunk 2 above)
Comment (NutharaNR) [reply features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:71]: Line break removed.

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx (hunk 4 above)
Comment (NutharaNR) [reply features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:95]: Line break removed.

File: features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+ * Configure the different sign on strategies for an application.
+ *
+ * @param props - Props injected to the component.
+ *
+ * @returns React element.
+ */
+export const SignOnMethodsWrapper: FunctionComponent<SignOnMethodsWrapperPropsInterface> = (
+    props: SignOnMethodsWrapperPropsInterface
+): ReactElement => {
+
```
[the remark this answers is not in the document]
Comment (NutharaNR) [reply features/admin.applications.v1/components/settings/sign-on-methods/components/sign-on-methods-wrapper.tsx:82]: Line break removed.

File: features/admin.applications.v1/components/settings/sign-on-methods/sign-on-methods.tsx (hunk 5 above)
Comment (NutharaNR) [reply features/admin.applications.v1/components/settings/sign-on-methods/sign-on-methods.tsx]: Added default `componentId`s to `sign-on-methods-wrapper.tsx` and `sign-on-methods.tsx`. 

The `componentId` passed from `edit-application.tsx` is forwarded to `sign-on-methods-core.tsx`  without any modifications. This ensures that the test suite failures are not caused due to changes in the `componentId`.

**Review by brionmario — APPROVED** [review]

**Pull request merged**